### ISSUE TITLE
hist_find_ndim crashes on images with an alpha channel

### ISSUE THREAD

**Issue by travisbell** [libvips/libvips#2634]
Hi John, I've been working on migrating IM workflows over to vips and everything has been going very smoothly, thanks so much for your hard work and dedication.

One issue I have stumbled on is that when trying to use `hist_find_ndim`, vips is crashing on images with transparency. On Mac OS, I get:

```
$ vips hist_find_ndim tmp/test2.png tmp/test2_histogram.png
Abort trap: 6
```

And on Linux, I get:

```
vips hist_find_ndim tmp/test2.png tmp/test2_histogram.png
*** stack smashing detected ***: terminated
Aborted
```

This is on the latest build:

```
vips -v
vips-8.12.2-Tue Jan 25 09:34:32 UTC 2022
```

Is there something I need to do different with `hist_find_ndim` for these images? I couldn't find anyone else talking about this...

**Comment by jcupitt** [comment]
Hi @travisbell, 

You're right, it doesn't check that bands <= 3, I'll change it.

As a workaround, you can take just the first three bands like this:

```
vips extract_band rgba.png rgb.png 0 --n 3
vips hist_find_ndim rgb.png histogram.png
```

Thanks for pointing out this dumbness.

**Comment by travisbell** [comment]
> As a workaround, you can take just the first three bands like this:
> 
> ```
> vips extract_band rgba.png rgb.png 0 --n 3
> vips hist_find_ndim rgb.png histogram.png
> ```

Perfect, this is working properly now.

> Thanks for pointing out this dumbness.

I choose to believe just an oversight and not dumbness! 😉

**Comment by jcupitt** [comment]
OK, fixed for 8.12.3, I credited you, I hope that's OK.
